casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -277,6 +279,29 @@ object Validate {
     } yield
       maybeCheckPoint match {
         case Some(_) => Right(Valid)
-        case None    => Left(InvalidTransaction)
+        case None    => Right(Valid) // TODO: Fix to Left(InvalidTransaction)
```

Comment:
@birchmd 

```
  // See [[/docs/casper/images/minimal_equivocation_neglect.png]] but cross out genesis block
  it should "not ignore equivocation blocks that are required for parents of proper nodes" in {
    val nodes   = HashSetCasperTestNode.network(validatorKeys.take(3), genesis)
    val deploys = (0 to 5).map(i => ProtoUtil.basicDeploy(i))

    // Creates a pair that constitutes equivocation blocks
    val Some(signedBlock1) = nodes(0).casperEff.deploy(deploys(0)) *> nodes(0).casperEff.createBlock
    val Some(signedBlock1Prime) = nodes(0).casperEff
      .deploy(deploys(1)) *> nodes(0).casperEff.createBlock

    nodes(1).casperEff.addBlock(signedBlock1)
    nodes(0).transportLayerEff.msgQueues(nodes(0).local).clear //nodes(0) misses this block
    nodes(2).transportLayerEff.msgQueues(nodes(2).local).clear //nodes(2) misses this block

    nodes(0).casperEff.addBlock(signedBlock1Prime)
    nodes(1).transportLayerEff.msgQueues(nodes(1).local).clear //nodes(1) misses this block
    nodes(2).receive()
```

For some reason when nodes(2) tries to receive that signedBlock1Prime and it validates the transactions, it says that it doesn't match the tuplespace hash stored in the block. I'm going to sleep on this now but can you think of a reason at quick glance?